### PR TITLE
Add documentation for scrapers

### PR DESCRIPTION
**Pull request by hancush**
## Description

Hark! A documentation site!

Succeeding build here: https://github.com/Metro-Records/scrapers-lametro/actions/runs/10622927449/job/29448271525

### Testing

- Review the documentation at https://metro-records.github.io/scrapers-lametro/ and confirm it's a good first draft.
- Try to run the Quarto docs locally following the instructions in the README and confirm you can preview the built site and make updates.

### CODE REVIEW

File: .github/workflows/publish_docs.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: quarto-dev/quarto-actions/setup@v2
+
+      - name: Render and Publish
+        if: steps.docs-changed.outputs.any_changed == 'true'
+        uses: quarto-dev/quarto-actions/publish@v2
+        with:
+          path: docs
+          target: gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (hancush) [inline .github/workflows/publish_docs.yml:34]: These steps are only run if there are changes in the `docs/` directory.

File: docker-compose.councilmatic.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    command: pupa update lametro
+    # Connect the scraper container to the app network
+    networks:
+      - app_net
+
+networks:
+  # Define connection to the app's Docker network
+  app_net:
+    name: la-metro-councilmatic_default
+    external: true
```
Comment (hancush) [inline docker-compose.councilmatic.yml:26]: Cc @xmedr

File: docs/debugging.qmd [hunk 3]
```diff
@@ -0,0 +1,224 @@
+---
```
Comment (hancush) [inline docs/debugging.qmd:1]: Migrated from the Councilmatic wiki.

File: docs/local-development.qmd [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+If you'd like to scrape data into a Councilmatic instance for easy viewing, first
+run your local instance of LA Metro Councilmatic as normal. 
+
+Then, in your local scraper repository, run your scrapes using the `docker-compose.councilmatic.yml` file:
+
+```bash
+docker compose -f docker-compose.councilmatic.yml run --rm scrapers
+```
+
+### Useful pupa commands
```
Comment (hancush) [inline docs/local-development.qmd:36]: Adapted from the Councilmatic wiki.

File: docs/debugging.qmd [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+- [`Metro-Records/la-metro-dashboard`](https://github.com/Metro-Records/la-metro-dashboard) is the Airflow app that schedules scrapes and the scripts that define them. If you need to change the scheduling of scrapes, create a PR against this repo.
+- [`opencivicdata/python-legistar-scraper/tree/master/legistar`](https://github.com/opencivicdata/python-legistar-scraper/tree/master/legistar) contains the `LegistarScraper` and `LegistarAPIScraper` variants, from which the Metro scrapers inherit. If you need to patch the Legistar scraping code, create a PR against this repo.
+- All scrapers depend on [the `pupa` framework](https://github.com/opencivicdata/pupa) for scraping and importing data using the OCD standard. In the unlikely event that you need to patch `pupa`, create a fork, then submit a PR against this repo.
+
+### Data is incorrect
+
+#### Legistar contains the wrong data
+
+Data issues can occur when the Legistar API or web interface displays the wrong information. This generally happens with Metro staff enters information that is incorrect.
```
Comment (xmedr) [inline docs/debugging.qmd]: might be a small typo. was it supposed to be something like:
> This generally happens \<when\> Metro staff enters information that is incorrect.

File: docs/debugging.qmd [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+## Inspecting data and logs
+
+### More on Airflow
+
+The Metro data pipeline, both scrapes and management commands to perform subsequent ETL, are scheduled and run by our Metro Airflow instance, located at [https://la-metro-dashboard-heroku-prod.datamade.us](https://la-metro-dashboard-heroku-prod.datamade.us). Consult the DataMade BitWarden for login credentials. (Search `metro-support@datamade.us` in our shared folder!)
+
+The dashboard lives on GitHub, [here](https://github.com/Metro-Records/la-metro-dashboard).
+
+Apache maintains thorough documentation of [core concepts](http://airflow.apache.org/docs/stable/concepts.html), as well as [navigating the UI](http://airflow.apache.org/docs/stable/ui.html). If you've never used Airflow before, these are great resources to start with!
```
Comment (xmedr) [inline docs/debugging.qmd:100]: this is super helpful!

File: docs/debugging.qmd [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+If you have reason to believe the scrape has failed, but you don't see an exception in Sentry, you can make double sure by consulting the logs for the scraping DAGs [in the dashboard](https://la-metro-dashboard-heroku-prod.datamade.us).
+
+#### Confirming the scrape ran
+
+To confirm the scrape ran, click the name of a DAG to pull up the tree view, which will display the status of the last 25 DAG runs.
+
+![windowed_bill_scraping DAG tree view](https://i.imgur.com/9NDdEpy.png)
+
+Note that the scraping DAGs employ [branch operators](https://airflow.apache.org/docs/stable/concepts.html?highlight=branch#branching) to determine what kind of scraping task to run. Be sure to look closely to verify that the task you're expecting is green (succeeded), not pink (skipped).
```
Comment (xmedr) [inline docs/debugging.qmd]: this link leads to the core concepts page without highlighting anything. would this link make sense?

https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#concepts-branching

File: docker-compose.councilmatic.yml (hunk 2 above)
Comment (xmedr) [reply docker-compose.councilmatic.yml:26]: ah this makes things way easier, ty!

File: docs/scrapers/events.qmd [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+order: 4
+---
+
+- Located in `lametro/events.py`
+- Creates events
+
+## Key issues
+
+- Metro streams audio in both English and Spanish. They cannot associated multiple
+broadcast link with one event in Legistar, so they create two nearly identical events that
```
Comment (xmedr) [inline docs/scrapers/events.qmd]: super small typo. is it supposed to be:
> They cannot associate multiple broadcast links with one event in Legistar, ...

**Review by xmedr — CHANGES_REQUESTED** [review]
This is really great! And the quarto docs run/update locally. Got some little comments below.

I also wanted to ask: I found that it took some time to figure out how some terms in Legistar mapped to Metro. For example, that "matter" equates to "board report", or a "body" refers to a "legislation" (I might be wrong about that). Would it be useful to have a small note of alternative names that certain items might be called in the individual scrapers sections?

File: docs/debugging.qmd [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+If you don't see an issue in Sentry, but you have reason to believe there was an error (or that our integration with Sentry is faulty), you can [follow our documentation for finding errors in the scraper logs](#inspect-the-scraper-logs).
+
+If there is not an error in Sentry or in the scraper logs, then the scrape ran. 
+
+#### Inaccurate timestamps in the Legistar API
+
+If scrapes are running, by far the most common source of issues is the scraper failing to capture changes to a bill or event. The root issue is that we rely on timestamps that should indicate an update to determine which bills and events to scrape. In reality, these timestamps do not always update when a change is made to an event or bill in Legistar. We have a couple of strategies to get around this:
+
+1. During [the Friday support window](https://github.com/datamade/la-metro-dashboard/blob/ac416e5e03f6a97fb9b0c6112093c679cefb0d1c/dags/constants.py#L41-L59), we scrape all events and bills at the top of every hour.
```
Comment (antidipyramid) [inline docs/debugging.qmd]: Should we remove references to the Friday support window?

**Review by antidipyramid — COMMENTED** [review]
Amazing @hancush! Just one question inline.

File: docs/debugging.qmd (hunk 9 above)
Comment (hancush) [reply docs/debugging.qmd]: Yes! Done.

**Comment by hancush** [comment]
@xmedr, @antidipyramid Thanks for the review, you two! Addressed your comments and published the changes manually: https://metro-records.github.io/scrapers-lametro/

Xavier, thanks for flagging synonyms! I've added those to each of the scraper pages, and I also added core concepts to the Jurisdiction scraper, since I found that really confusing when I got started. I also added links to the Open Civic Data documentation where relevant. They're really useful for understanding why data is structured the way it is.

Let me know if this looks good to merge!

**Review by xmedr — APPROVED** [review]
These are hugely helpful, thank you! Looks good to me

**Pull request merged**